packages/nextjs/app/builders/page.tsx

```diff
@@ -0,0 +1,73 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Image from "next/image";
+import Link from "next/link";
+import makeBlockie from "ethereum-blockies-base64";
+import type { NextPage } from "next";
+import { GlobeAltIcon } from "@heroicons/react/24/solid";
+import { useScaffoldEventHistory } from "~~/hooks/scaffold-eth";
+
+const BuildersPage: NextPage = () => {
+  const [builders, setBuilders] = useState<string[]>([]);
+
+  const { data: checkedInEvents, isLoading } = useScaffoldEventHistory({
+    contractName: "BatchRegistry",
+    eventName: "CheckedIn",
+    fromBlock: 118484927n,
+    watch: true,
+  });
+
+  useEffect(() => {
+    if (!isLoading && checkedInEvents) {
+      const builderAddresses = checkedInEvents
+        .map(event => event.args.builder as string | undefined)
+        .filter((builder): builder is string => !!builder);
+      const uniqueBuilders = Array.from(new Set(builderAddresses));
```

Comment:
You don't need `useEffect` and an extra `builders`  state. 

You could move this out of `useEffect` and use `uniqueBuilders` directly while rendering on UI